### ISSUE TITLE
Training does not start when using multi GPU

### ISSUE THREAD

**Issue by shan18** [mozilla/DeepSpeech#2356]
- **Have I written custom code (as opposed to running examples on an unmodified clone of the repository)**: No
- **OS Platform and Distribution**: Linux Ubuntu 18.04
- **TensorFlow installed from**: upstream TensorFlow
- **TensorFlow version**: 1.14.0
- **Python version**: 3.6.8
- **CUDA/cuDNN version**: 10.0 / 7.6
- **GPU model and memory**: 2 x GTX 1080 Ti (Each with 12 GB memory)
- **Exact command to reproduce**:
```[python]
python3 DeepSpeech.py --n_hidden 2048 --epoch 100 \
--train_files data/train.csv \
--dev_files data/dev.csv \
--test_files data/test.csv \
--checkpoint_dir ~/checkpoints \
--export_dir ~/checkpoints/export \
--train_batch_size 24 \
--dev_batch_size 24 \
--test_batch_size 48 \
--learning_rate 0.0001 \
--dropout_rate 0.15 \
--lm_alpha 0.75 \
--lm_beta 1.85 \
--alphabet_config_path data/alphabet.txt \
--beam_width 1024
```
I was trying to train the model on the Librispeech dataset. When I run the code on a single GPU, it works fine. But when I try to run the code on multiple GPUs, the model just stops at this step:  
`Epoch 0 |   Training | Elapsed Time: 0:00:00 | Steps: 0 | Loss: 0.000000`  
And when I check the memory usage using `nvidia-smi`, it shows that both the GPUs have 10 GB of memory full due to this process.

On killing the process, the process remains in the background and it does not seem to stop even when I try to kill it with `kill -9 <pid>`. The only option left after this is to reboot the machine in order to kill the process.

Note: To make the model run on a single GPU, I just make one of the GPUs as discoverable and for multiple GPUs, I just run the code as it is.

**Comment by kdavis-mozilla** [comment]
What are the env variables you run the multi-gpu code under?

**Comment by reuben** [comment]
To be clear, we exercise this code daily and it works fine with one or several GPUs.

**Comment by shan18** [comment]
Okay, I'll check my installation. Thank you for the help.

**Comment by lissyx** [comment]
>  `Epoch 0 | Training | Elapsed Time: 0:00:00 | Steps: 0 | Loss: 0.000000`

Sharing more logs could have helped us help you.

**Comment by shan18** [comment]
@lissyx  I don't have any more logs. The training just stops at this point. I waited for 45 mins for it to start, it didn't. Then when I ran it on a single GPU, it worked as usual.

**Comment by lissyx** [comment]
> @lissyx I don't have any more logs. The training just stops at this point. I waited for 45 mins for it to start, it didn't. Then when I ran it on a single GPU, it worked as usual.

You should have or force having more logs. `TF_CPP_MIN_LOG_LEVEL` does that, and we have `--log_level` to expose that.

**Comment by shan18** [comment]
Okay, I'll try once more and report back the logs.

**Comment by biligunb** [comment]
> Okay, I'll try once more and report back the logs.

Probably a problem with your dataset.
Maybe with the generated `.csv`. (I had that one too)
Can you show the csv generation process logs. (I think you might find the problem & solution there)

**Comment by MuruganR96** [comment]
i also facing same problem. how to resolve this issue.? @shan18 @lissyx @reuben 

```
CUDA_VISIBLE_DEVICES=1,2,3  python3 DeepSpeech.py --n_hidden 2048 --checkpoint_dir  /media/use/  --dropout_rate 0.12 --epochs 50 --train_files /media/user1/storage-1/suneel/stt/data/train1.csv  --dev_files  /media/user1/storage-1/suneel/stt/data/dev1.csv --test_files  /media/user1/storage-1/suneel/stt/data/test1.csv --train_batch_size 1 --test_batch_size 1 --dev_batch_size 1 --learning_rate 0.0001 --export_dir /media/user1/storage-1/suneel/stt/DeepSpeech/model_files_dir/ --one_shot_infer /media/user1/storage-1/suneel/stt/checkpoints/testing/6.wav --es_steps 8 2>&1 | tee ds.log
WARNING:tensorflow:From /media/user1/storage-1/suneel/stt/stt_env/lib/python3.6/site-packages/tensorflow/python/data/ops/dataset_ops.py:429: py_func (from tensorflow.python.ops.script_ops) is deprecated and will be removed in a future version.
Instructions for updating:
tf.py_func is deprecated in TF V2. Instead, use
    tf.py_function, which takes a python function which manipulates tf eager
    tensors instead of numpy arrays. It's easy to convert a tf eager tensor to
    an ndarray (just call tensor.numpy()) but having access to eager tensors
    means tf.py_functions can use accelerators such as GPUs as well as
    being differentiable using a gradient tape.
    
WARNING:tensorflow:From /media/user1/storage-1/suneel/stt/stt_env/lib/python3.6/site-packages/tensorflow/python/data/ops/iterator_ops.py:358: colocate_with (from tensorflow.python.framework.ops) is deprecated and will be removed in a future version.
Instructions for updating:
Colocations handled automatically by placer.
WARNING:tensorflow:From /media/user1/storage-1/suneel/stt/stt_env/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/lstm_ops.py:696: to_int64 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.cast instead.
WARNING:tensorflow:From /media/user1/storage-1/suneel/stt/stt_env/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1266: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
I Restored variables from most recent checkpoint at /media/user1/storage-1/suneel/stt/DeepSpeech/ckpt_file_v0.5.1/deepspeech-0.5.1-checkpoint/model.v0.5.1, step 467356
I STARTING Optimization
```

automatically training killed and no more error log.

**Comment by lissyx** [comment]
You have a bogus wav file..

**Comment by MuruganR96** [comment]
how to resolve @lissyx . 

we need to change the sampling rate, using sox

```
from subprocess import Popen, PIPE, STDOUT
soxCall = '/usr/local/bin/sox ' + infileName + \
                 ' ' + outfileName + ' rate 16k'
p = Popen(soxCall, shell=True, stdin=PIPE, stdout=PIPE, stderr=STDOUT, close_fds=True) 
```

is the enough to resolve this issue @lissyx . thank you:)

**Comment by lissyx** [comment]
> how to resolve @lissyx sir.
> thank you sir:)

Find and remove it ?

**Comment by MuruganR96** [comment]
Is it empty wavfile. Or corrupted wav file.? 
Mm thank you sir.

**Comment by lissyx** [comment]
> Is it empty wavfile. Or corrupted wav file.?
> Mm thank you sir.

I don't have your data, I can't tell you.

> **Invalid argument: Header mismatch: Expected RIFF but found ����**
